ra2ce/analyses/indirect/analyses_indirect.py

```diff
@@ -995,26 +1013,59 @@ def multi_link_isolated_locations(self, graph, analysis, crs=4326):
                 crs=crs,
             )
 
-            # remove the largest (main) graph
+            # remove the largest (main) graph, leaving isolated network
             results.sort_values(by="count", ascending=False, inplace=True)
             results = results.loc[results["count"] != results["count"].max()]
 
             # reproject the datasets to be able to make a buffer in meters
             results = results.set_crs(crs=crs)
             results.to_crs(crs=nearest_utm, inplace=True)
 
+            # process network to get output
+            # get directly disrupted roads
+            results_flooded_roads = gpd.GeoDataFrame(
+                {
+                    "geometry": [e[-1]["geometry"] for e in edges_remove],
+                    "edge_fid": [
+                        f"{e[-1]['node_A']}_{e[-1]['node_B']}" for e in edges_remove
+                    ],
+                },
+                crs=crs,
+            )
+            results_flooded_roads.to_crs(crs=nearest_utm, inplace=True)
+            results_flooded_roads.geometry = results_flooded_roads.geometry.buffer(
+                analysis["buffer_meters"],
+                cap_style=3,
+            )
+            results_flooded_roads[f"i_type_{hazard_name[:-3]}"] = "flooded"
+
+            # get indirect isolated roads
             results_buffered = results.copy()
             results_buffered.geometry = results_buffered.geometry.buffer(
-                analysis["buffer_meters"]
+                analysis["buffer_meters"],
+                cap_style=3,
             )
+            results_buffered[f"i_type_{hazard_name[:-3]}"] = "isolated"
+
+            # merge for output, set original crs
+            results_buffered = gpd.GeoDataFrame(
+                pd.concat([results_buffered, results_flooded_roads])
+            ).to_crs(crs=crs)
+
+            # Save the output
             results_buffered.to_file(
```

Comment:
the whole `results_buffered` could be in a different method `get_results_buffered_flooded_roads(...)` so that the code becomes more readable and maintainable.

Reply:
A new function called buffer_geometry is created. Because it is a very generic function, I put in network_utils.py. Please suggest if it fits better somewhere else.